tests/Doctrine/Annotation/TokenTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Doctrine\Annotation;
+
+use Doctrine\Common\Annotations\DocLexer;
+use PhpCsFixer\Doctrine\Annotation\Token;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Doctrine\Annotation\Token
+ */
+final class TokenTest extends TestCase
+{
+    public function testDefaults()
+    {
+        $token = new Token();
+
+        $this->assertSame(DocLexer::T_NONE, $token->getType());
+        $this->assertSame('', $token->getContent());
+    }
+
+    public function testConstructorSetsValues()
+    {
+        $type = 42;
+        $content = 'questionable';
+
+        $token = new Token(
+            $type,
+            $content
+        );
+
+        $this->assertSame($type, $token->getType());
+        $this->assertSame($content, $token->getContent());
+    }
+
+    public function testCanModifyType()
+    {
+        $type = 42;
+
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertSame($type, $token->getType());
+    }
+
+    /**
+     * @dataProvider providerIsType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsTrue($type, $types)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertTrue($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsType()
+    {
+        return array(
+            'same' => array(
+                42,
+                42,
+            ),
+            'array-containing-same' => array(
+                42,
+                array(
+                    42,
+                    9001,
+                ),
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerIsNotType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsFalse($types, $type)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertFalse($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsNotType()
+    {
+        return array(
+            'equal-but-not-same' => array(
```

Comment:
This cases are violating the test and the tested methods API as they both expect `int|int[]`. I would either add validation in the tested method (e.g. throwing an exception when a string is passed instead of an integer) or remove these test cases.

Anyway, can you please add cases with different integers?